### DISCUSSION TITLE
Docker images

### DISCUSSION THREAD

**Discussion by FerronN** [dani-garcia/vaultwarden#2562] [General]
If I'm correct at the moment new docker images are only build when there is a new release.
I think there should be a new automated build every week, so all possible security updates are included the docker image.

**Comment by stefan0xC** [comment]
> If I'm correct at the moment new docker images are only build when there is a new release.

The [**testing** images](https://hub.docker.com/r/vaultwarden/server/tags?page=1&name=testing) are built on new commits to the main branch, so you are incorrect.
The difference between latest and testing has been explained in the pinned issue #1180

**Comment by FerronN** [comment]
@stefan0xC I don't agree, because we're talking about two different thing. I talk about the docker image itself and its dependencies.
Depending on which image you're using it has a specific runtime container with certain apk or apt packages. The keeps this up to date it would be nice to have a new image every week or so. Not with the latest commits, but with de latest runtime image and dependencies.

**Comment by BlackDex** [comment]
While i kinda understand the reason for this, i just checked the `apt update` of the `latest` images and there arn't really updates that need to be installed i think.

Also, if i look at the runtime images, the Debian one was last updated 23 days ago, and the Alpine images we use 14 days ago.
So, building them every week would probably be to much. And, i also do not think it will be that useful if there are no packages updated which have any linkage to Vaultwarden it self, like `dpkg` or `apk` packages them selfs.

If you want to have the most up-to-date version i would suggest to use the `testing` tagged images, those are build on every commit using the images available at that specific point in time.

It is however something to keep in mind maybe for the future if there hasn't been an update for a long time for example.
Or maybe we need to think about an updated image every two weeks, or only when the runtime images have been updated for example.

**Comment by FerronN** [comment]
I agree when the time between releases is not to long. The chance for important security updates is then not that big. But there were 113 days between v1.24 and v1.25. When I thought about posting this v1.25 was released. But 113 without image update is very long.
